src/ng/directive/select.js

```diff
@@ -70,7 +70,9 @@ var SelectController =
   };
 
   self.unselectEmptyOption = function() {
-    if (self.emptyOption) {
+    // Empty options may be comment nodes if they have been removed by
+    // transclude directives such as `ngIf`
+    if (self.emptyOption && self.emptyOption[0].nodeType === NODE_TYPE_ELEMENT) {
```

Comment:
That's right. That means the whole adding / removing "selected" attribute is currently unsound when ngIf / a transclusion directive is used. 
and it's not as easy as looking for the first non-comment element, because self.emptyOption is sometimes *only* the comment